src/utils/validations/PayloadValidation.ts

```diff
@@ -26,16 +29,31 @@ const featureFlagValidation = (validationSchema: ObjectSchema, payload: ITechRec
   }
 };
 
-export const validatePayload = (payload: ITechRecord, validateEntireRecord: boolean = true) => {
-  const isTankOrBattery = checkIfTankOrBattery(payload);
+/**
+ * common validation function for both HGVs and Trailers.
+ * @param payload payload which needs to be validated
+ * @param options validation configurations
+ * @param isCreate true for create request and false for update request
+ */
+export const validateHGVOrTrailer =(payload: HgvTechRecord | TrlTechRecord, options: any , isCreate: boolean) => {
+  const isTankOrBattery: boolean = checkIfTankOrBattery(payload);
+  options = {...options, context: {isTankOrBattery}};
+  const validationSchema = payload.vehicleType === VEHICLE_TYPE.HGV? hgvValidation:trlValidation;
+  const validationResult = featureFlagValidation(validationSchema, payload, isCreate, options);
+  return handleValidationResult(validationResult);
+};
+
+// will be removed
+export const validatePayload = (payload: TechRecord, validateEntireRecord: boolean = true) => {
```

Comment:
Do you still need this validator since you created the validateHGVOrTrailer method?

Reply:
removed.